### DISCUSSION TITLE
How is set reward fee to relaying accounts ?

### DISCUSSION THREAD

**Discussion by AlejoLovallo** [informalsystems/hermes#2220] [Q&A] [answered]
I already have set up and run a relayer between an ignite scaffolded blockchain and cosmos-hub. I´ve provided two accounts to pay for the transaction fees and I have also made another test by setting the fee_granter option.  So far so good. 

My question is where (if it is possible), or how to set the reward fee that a user that wants to use the relayer should pay?

**Comment by adizere** [comment, marked as answer]
Hi,

The relayer fees are not ready for use yet. The last release candidate https://github.com/cosmos/ibc-go/releases/tag/v4.0.0-rc1 was just published this week and the final release will likely land in 1-2 weeks.

Hermes relayer supports this feature already so you can play with it https://github.com/informalsystems/ibc-rs/pull/1981, but it is not yet merged in master until the final tag is done in the ibc-go repository.

Good resources for you to understand relayer fees are here:
- https://interchain-io.medium.com/ibc-relaying-as-a-service-the-in-protocol-incentivization-story-9922c7b953f0
- https://ibc.cosmos.network/main/middleware/ics29-fee/overview.html

>  how to set the reward fee that a user that wants to use the relayer should pay?

The flow is: 
1. user locks relayer fees when they submit their transaction, just like they do for normal tx gas fees
2. the relayer who relays that transaction will "win" those locked relayer fees

I actually don't know what the user UI looks like exactly, tagging @womensrights as she might know.
